File: src/main/scala/com/badlogic/gdx/graphics/GL20.scala
/*******************************************************************************
  * Copyright 2011 See AUTHORS file.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *   http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
  ******************************************************************************/
package com.badlogic.gdx.graphics

import scala.scalajs.js
import scala.scalajs.js.annotation.JSName

/** Interface wrapping all the methods of OpenGL ES 2.0
  * @author mzechner */
@JSName("gdx.graphics.GL20")
@js.native
object GL20 extends js.Object {
  var GL_COLOR_BUFFER_BIT: Double = js.native
  var GL_NEAREST: Double = js.native
  var GL_LINEAR: Double = js.native
  var GL_NEAREST_MIPMAP_NEAREST: Double = js.native
  var GL_LINEAR_MIPMAP_NEAREST: Double = js.native
  var GL_NEAREST_MIPMAP_LINEAR: Double = js.native
  var GL_LINEAR_MIPMAP_LINEAR: Double = js.native
}

@JSName("gdx.graphics.GL20")
@js.native
class GL20 extends js.Object {
  def glClearColor(red: Double, green: Double, blue: Double, alpha: Double): Unit = js.native
  def glClear(mask: Double): Unit = js.native
}

File: libgdx/src/gdx/graphics/GL20.js
gdx.graphics.GL20 = (function() {

    var Gdx = gdx.Gdx;
    /**
     * @JSName("gdx.graphics.GL20")
     */
    class GL20 {
        glClearColor(red, green, blue, alpha) {
            let hexColor = ((1 << 24) + (red*255 << 16) + (green*255 << 8) + blue*255)//.toString(16).substr(1);
            Gdx._renderer.backgroundColor = hexColor;
        }
        glClear(mask) {
        }
    }

    GL20.GL_COLOR_BUFFER_BIT = 0x00004000;
    GL20.GL_NEAREST = 0x2600;
    GL20.GL_LINEAR = 0x2601;
    GL20.GL_LINEAR_MIPMAP_LINEAR = 0x2703;
    GL20.GL_NEAREST_MIPMAP_NEAREST = 0x2700;
    GL20.GL_LINEAR_MIPMAP_NEAREST = 0x2701;
    GL20.GL_NEAREST_MIPMAP_LINEAR = 0x2702;
    GL20.GL_LINEAR_MIPMAP_LINEAR = 0x2703;

    return GL20;

}());
